platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,852 @@
+* Perform a Signature verification via EVM code of a signature produced with the library.
+
+* Some ideas:
+* [Cryptographic-Algorithm-Validation-Program](https://csrc.nist.gov/CSRC/media//Projects/Cryptographic-Algorithm-Validation-Program/documents/dss/ECDSAVS.pdf):
+* https://csrc.nist.gov/CSRC/media/Events/lightweight-cryptography-workshop-2019/documents/papers/systematic-testing-of-lightweight-crypto-lwc2019.pdf
+
+The proposal is to move forward with the implementation while creating a work group to discuss and collect ideas for testing and validating these features.
+
+### Unit Tests
+One of the components that can be unit-tested is native support. SPI loading and failing. JNI wrapping.
+Limitations: native-support-library tests must be executed in different environments to provide comprehensive code validation.
+
+### Performance Testing
+JMH benchmarks should be provided for signature generation and aggregation.
+TssMessage validation, and public key aggregation.
+
+We should know that variance (min, max, average, median) time it takes to do the following:
+
+Genesis for a 4, 10, 20, 40, and 100 participants.
+Rekeying for a 4, 10, 20, 40, and 100 participants.
+Shares Amounts distributed randomly: 200, 500, 1000, 2000, 5000
+we need to test each configuration 100 times to get confident mins, maxes, and a good average and median
+
+## Security Audit
+After this proposal is accepted we will invite security team to define the necessary steps for auditing the code.
+
+## Alternatives Considered
+- As mentioned before, Java Foreign Function Interface is still immature so it was not considered for this proposal.
+- Using JCA through Bouncy Castle implementation was analyzed. BC provides support for EC-Curves, but it does not support Pairings which are required for BLS signatures.
+  [beacon-chain](https://github.com/harmony-dev/beacon-chain-java/tree/master) implemented the support using bouncy castle + [Milagro](https://incubator.apache.org/projects/milagro.html) 
+but Milagro project is reported to have little coverage, not audited. Milagro podling has been retired on 2024-03-21.
+
+
+## Implementation and Delivery Plan by stages
+**Stage 1**
+* Preconditions:
+  * hedera-cryptography repository.
+  * Gradle multilanguage module with rust compilation plugin.
+  * CI/CD pipelines for building artifacts in hedera-cryptography.
+* Define the Test plan.
+* Define a security plan.
+* Implementation of native-support library.
+* Implementation of Pairings API using JNI, arkworks, and alt-bn128.
+* Execute Test Plan and validation.
+* Execute Security Audits.
+* Implementation of EC-key utility.
+
+**Stage 2**
+* Preconditions:
+  * CI/CD pipelines to reference built artifacts in hedera-cryptography from hedera-services.
+  * Implementation of the public interface for the TSS library.
```

Comment:
I assume this means just checking in the API that the consensus node will use. Could this be done in stage 1 so it can be used by the team working on TSS Block Signing? I do not consider this a material change.

Reply:
Yes, it makes sense. I was optimizing stage 1 to deliver the capacity to create the signatures for DevOps. 
But that is not the only front.
Also I realize I removed the implementation of the **swirlds-crypto-signatures** from the implementation plan by mistake.

I will create a follow-up PR for this proposal with the updated implementation plan.

Reply:
Sorry, misread `I do not consider this a material change`
Won't create a follow-up PR based on that.